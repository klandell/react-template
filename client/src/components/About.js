import React, { Component } from 'react';
import { string } from 'prop-types';
import style_ from 'Sass/containers/About.scss';

export default class About extends Component {
    static propTypes = {
        param1: string,
        param2: string,
    }

    static defaultProps = {
        param1: '',
        param2: '',
    }

    render() {
        const { param1, param2 } = this.props;
        return (
            <div className="about-wrapper">
                <h1>About</h1>
                {param1}
                {param2}
            </div>
        );
    }
}
