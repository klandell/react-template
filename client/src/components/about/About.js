import React, { Component } from 'react';
import { string } from 'prop-types';
import style_ from './About.scss';

export default class About extends Component {
    static propTypes = {
        param1: string,
        param2: string,
    }

    static defaultProps = {
        param1: '',
        param2: '',
    }

    renderParam(value) {
        return value ? <h1>{value}</h1> : null;
    }

    render() {
        const { param1, param2 } = this.props;
        return (
            <div className="about">
                <h1>About</h1>
                {this.renderParam(param1)}
                {this.renderParam(param2)}
            </div>
        );
    }
}
