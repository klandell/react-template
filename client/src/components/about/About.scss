@import '~Theme/colors';

.about {
    color: $cobalt;
}
