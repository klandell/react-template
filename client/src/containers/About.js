import React from 'react';
import style_ from 'Sass/containers/About.scss';

export default class About extends React.Component {
    renderParam(param) {
        const value = this.props.match.params[param];
        return value ? <h1>{value}</h1> : null;
    }

    render() {
        const param1 = this.renderParam('param1');
        const param2 = this.renderParam('param2');

        return (
            <div className="about-wrapper">
                <h1>About</h1>
                {param1}
                {param2}
            </div>
        );
    }
}
