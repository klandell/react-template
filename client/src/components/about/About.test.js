import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import About from './About';

test('About component should be able to render no path parameters.', () => {
    const component = shallow(<About />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
});

test('About component should be able to render one path parameter.', () => {
    const component = shallow(<About param1="Hello" />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
});

test('About component should be able to render two path parameters.', () => {
    const component = shallow(<About param1="Hello" param2="World" />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
});
