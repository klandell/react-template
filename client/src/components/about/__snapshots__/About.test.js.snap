// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component should be able to render no path parameters. 1`] = `
<div
  className="about"
>
  <h1>
    About
  </h1>
</div>
`;

exports[`About component should be able to render one path parameter. 1`] = `
<div
  className="about"
>
  <h1>
    About
  </h1>
  <h1>
    Hello
  </h1>
</div>
`;

exports[`About component should be able to render two path parameters. 1`] = `
<div
  className="about"
>
  <h1>
    About
  </h1>
  <h1>
    Hello
  </h1>
  <h1>
    World
  </h1>
</div>
`;
